## France or Germany? Online retail

A retail company asks us to analyze the sales for the years 2021 and 2022 to understand where they should invest more, whether in the branch in France or Germany. We proceed to import the data from a csv file.

In [1]:
#import pandas and data form cvs
import pandas as pd

In [7]:
pd.read_csv('On_line_retail.csv')

InvoiceNo StockCode                      Description  Quantity  \
0        536370     22728        ALARM CLOCK BAKELIKE PINK        24   
1        536370     22727        ALARM CLOCK BAKELIKE RED         24   
2        536370     22726       ALARM CLOCK BAKELIKE GREEN        12   
3        536370     21724  PANDA AND BUNNIES STICKER SHEET        12   
4        536370     21883                 STARS GIFT TAPE         24   
...         ...       ...                              ...       ...   
18047    581587     22613      PACK OF 20 SPACEBOY NAPKINS        12   
18048    581587     22899     CHILDREN'S APRON DOLLY GIRL          6   
18049    581587     23254    CHILDRENS CUTLERY DOLLY GIRL          4   
18050    581587     23255  CHILDRENS CUTLERY CIRCUS PARADE         4   
18051    581587     22138    BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate UnitPrice  CustomerID Country  
0      2021-12-01 08:45:00      3,75     12583.0  France  
1      2021-12-01 08:45:00      3,75     12583.0  France  
2      2021-12-01 08:45:00      3,75     12583.0  France  
3      2021-12-01 08:45:00      0,85     12583.0  France  
4      2021-12-01 08:45:00      0,65     12583.0  France  
...                    ...       ...         ...     ...  
18047  2022-12-09 12:50:00      0,85     12680.0  France  
18048  2022-12-09 12:50:00       2,1     12680.0  France  
18049  2022-12-09 12:50:00      4,15     12680.0  France  
18050  2022-12-09 12:50:00      4,15     12680.0  France  
18051  2022-12-09 12:50:00      4,95     12680.0  France  

[18052 rows x 8 columns]

In [6]:
import os
os.chdir('/Users/OBPC/Downloads')

In [9]:
df =pd.read_csv('On_line_retail.csv')

In [14]:
df[df.notnull()]

InvoiceNo StockCode                      Description  Quantity  \
0        536370     22728        ALARM CLOCK BAKELIKE PINK        24   
1        536370     22727        ALARM CLOCK BAKELIKE RED         24   
2        536370     22726       ALARM CLOCK BAKELIKE GREEN        12   
3        536370     21724  PANDA AND BUNNIES STICKER SHEET        12   
4        536370     21883                 STARS GIFT TAPE         24   
...         ...       ...                              ...       ...   
18047    581587     22613      PACK OF 20 SPACEBOY NAPKINS        12   
18048    581587     22899     CHILDREN'S APRON DOLLY GIRL          6   
18049    581587     23254    CHILDRENS CUTLERY DOLLY GIRL          4   
18050    581587     23255  CHILDRENS CUTLERY CIRCUS PARADE         4   
18051    581587     22138    BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate UnitPrice  CustomerID Country  
0      2021-12-01 08:45:00      3,75     12583.0  France  
1      2021-12-01 08:45:00      3,75     12583.0  France  
2      2021-12-01 08:45:00      3,75     12583.0  France  
3      2021-12-01 08:45:00      0,85     12583.0  France  
4      2021-12-01 08:45:00      0,65     12583.0  France  
...                    ...       ...         ...     ...  
18047  2022-12-09 12:50:00      0,85     12680.0  France  
18048  2022-12-09 12:50:00       2,1     12680.0  France  
18049  2022-12-09 12:50:00      4,15     12680.0  France  
18050  2022-12-09 12:50:00      4,15     12680.0  France  
18051  2022-12-09 12:50:00      4,95     12680.0  France  

[18052 rows x 8 columns]

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18052 entries, 0 to 18051
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   InvoiceNo    18052 non-null  object 
 1   StockCode    18052 non-null  object 
 2   Description  18052 non-null  object 
 3   Quantity     18052 non-null  int64  
 4   InvoiceDate  18052 non-null  object 
 5   UnitPrice    18052 non-null  object 
 6   CustomerID   17986 non-null  float64
 7   Country      18052 non-null  object 
dtypes: float64(1), int64(1), object(6)
memory usage: 1.1+ MB


We are detecting a few issues in de df: change cutomer id, removing the float, and change type to int. As well customer id probably has some nulls. Quantity needs to be float type.

In [17]:
df.describe()

Quantity    CustomerID
count  18052.000000  17986.000000
mean      12.626191  12661.178528
std       19.634816    294.710582
min     -288.000000  12413.000000
25%        5.000000  12523.000000
50%       10.000000  12625.000000
75%       12.000000  12683.000000
max      912.000000  14335.000000

We found another issue: the min of quantity shouledn be negative, we will explore this sooner.

### Explorations of nulls in CustomerID

In [20]:
df[df.CustomerID.isnull()]

InvoiceNo StockCode                         Description  Quantity  \
7525     556924     82552                 WASHROOM METAL SIGN         1   
7526     556924     21165       BEWARE OF THE CAT METAL SIGN          1   
7527     556924    47503A  ASS FLORAL PRINT MULTI SCREWDRIVER         1   
7528     556924     23154            SET OF 4 JAM JAR MAGNETS         3   
7529     556924     23156       SET OF 5 MINI GROCERY MAGNETS         3   
...         ...       ...                                 ...       ...   
15452    575759     22617          BAKING SET SPACEBOY DESIGN         1   
15453    575759     84836         ZINC METAL HEART DECORATION        24   
15454    575759     21811             HANGING HEART WITH BELL        48   
15455    575759     21822   GLITTER CHRISTMAS TREE WITH BELLS        12   
15456    575759     21034        REX CASH+CARRY JUMBO SHOPPER         1   

               InvoiceDate UnitPrice  CustomerID Country  
7525   2022-06-15 14:14:00      1,45         NaN  France  
7526   2022-06-15 14:14:00      1,69         NaN  France  
7527   2022-06-15 14:14:00      1,25         NaN  France  
7528   2022-06-15 14:14:00      2,08         NaN  France  
7529   2022-06-15 14:14:00      2,08         NaN  France  
...                    ...       ...         ...     ...  
15452  2022-11-11 10:50:00      4,95         NaN  France  
15453  2022-11-11 10:50:00      1,25         NaN  France  
15454  2022-11-11 10:50:00      0,39         NaN  France  
15455  2022-11-11 10:50:00      0,79         NaN  France  
15456  2022-11-11 10:50:00      0,95         NaN  France  

[66 rows x 8 columns]

66 rows has NaN customers ID

In [22]:
#we need to change unitprice type
df.UnitPrice.dtype

dtype('O')

In [24]:
#because is "object" we need to update it to type float
df.UnitPrice = df.UnitPrice.apply( lambda x: float(x.replace(',','.'))) 

We can check "describe" to have a quick look and check negative values

In [26]:
df.describe()

Quantity     UnitPrice    CustomerID
count  18052.000000  18052.000000  17986.000000
mean      12.626191      4.470307  12661.178528
std       19.634816     56.311410    294.710582
min     -288.000000      0.000000  12413.000000
25%        5.000000      1.250000  12523.000000
50%       10.000000      1.950000  12625.000000
75%       12.000000      3.750000  12683.000000
max      912.000000   4161.060000  14335.000000

In [28]:
#check negative values
df[df.Quantity<0]

InvoiceNo StockCode                          Description  Quantity  \
35      C536548     22244           3 HOOK HANGER MAGIC GARDEN        -4   
36      C536548     22242        5 HOOK HANGER MAGIC TOADSTOOL        -5   
37      C536548     20914  SET/5 RED RETROSPOT LID GLASS BOWLS        -1   
38      C536548     22892    SET OF SALT AND PEPPER TOADSTOOLS        -7   
39      C536548     22654                   DELUXE SEWING KIT         -1   
...         ...       ...                                  ...       ...   
17950   C581409     82484    WOOD BLACK BOARD ANT WHITE FINISH        -1   
17951   C581409     82482    WOODEN PICTURE FRAME WHITE FINISH        -1   
17952   C581409     22173   METAL 4 HOOK HANGER FRENCH CHATEAU        -2   
17953   C581409    85199L  LARGE HANGING IVORY & RED WOOD BIRD        -1   
17954   C581409     85127   SMALL SQUARE CUT GLASS CANDLESTICK        -5   

               InvoiceDate  UnitPrice  CustomerID  Country  
35     2021-12-01 14:33:00       1.95     12472.0  Germany  
36     2021-12-01 14:33:00       1.65     12472.0  Germany  
37     2021-12-01 14:33:00       2.95     12472.0  Germany  
38     2021-12-01 14:33:00       1.25     12472.0  Germany  
39     2021-12-01 14:33:00       5.95     12472.0  Germany  
...                    ...        ...         ...      ...  
17950  2022-12-08 14:08:00       7.95     12476.0  Germany  
17951  2022-12-08 14:08:00       2.95     12476.0  Germany  
17952  2022-12-08 14:08:00       3.29     12476.0  Germany  
17953  2022-12-08 14:08:00       0.65     12476.0  Germany  
17954  2022-12-08 14:08:00       4.95     12476.0  Germany  

[602 rows x 8 columns]

602 rows has negative quantities and invoiceNo starts with "C", we will check if this are devolutions

In [29]:
df[(df.InvoiceNo.str.contains('C'))& (df.Quantity<0)]

InvoiceNo StockCode                          Description  Quantity  \
35      C536548     22244           3 HOOK HANGER MAGIC GARDEN        -4   
36      C536548     22242        5 HOOK HANGER MAGIC TOADSTOOL        -5   
37      C536548     20914  SET/5 RED RETROSPOT LID GLASS BOWLS        -1   
38      C536548     22892    SET OF SALT AND PEPPER TOADSTOOLS        -7   
39      C536548     22654                   DELUXE SEWING KIT         -1   
...         ...       ...                                  ...       ...   
17950   C581409     82484    WOOD BLACK BOARD ANT WHITE FINISH        -1   
17951   C581409     82482    WOODEN PICTURE FRAME WHITE FINISH        -1   
17952   C581409     22173   METAL 4 HOOK HANGER FRENCH CHATEAU        -2   
17953   C581409    85199L  LARGE HANGING IVORY & RED WOOD BIRD        -1   
17954   C581409     85127   SMALL SQUARE CUT GLASS CANDLESTICK        -5   

               InvoiceDate  UnitPrice  CustomerID  Country  
35     2021-12-01 14:33:00       1.95     12472.0  Germany  
36     2021-12-01 14:33:00       1.65     12472.0  Germany  
37     2021-12-01 14:33:00       2.95     12472.0  Germany  
38     2021-12-01 14:33:00       1.25     12472.0  Germany  
39     2021-12-01 14:33:00       5.95     12472.0  Germany  
...                    ...        ...         ...      ...  
17950  2022-12-08 14:08:00       7.95     12476.0  Germany  
17951  2022-12-08 14:08:00       2.95     12476.0  Germany  
17952  2022-12-08 14:08:00       3.29     12476.0  Germany  
17953  2022-12-08 14:08:00       0.65     12476.0  Germany  
17954  2022-12-08 14:08:00       4.95     12476.0  Germany  

[602 rows x 8 columns]

In [30]:
df[(df.InvoiceNo.str.contains('C'))& (df.Quantity>0)]

Empty DataFrame
Columns: [InvoiceNo, StockCode, Description, Quantity, InvoiceDate, UnitPrice, CustomerID, Country]
Index: []

We conclude the invoices staring with C are devolutions.

In [31]:
df[df.CustomerID == 12472.0]

InvoiceNo StockCode                          Description  Quantity  \
35      C536548     22244           3 HOOK HANGER MAGIC GARDEN        -4   
36      C536548     22242        5 HOOK HANGER MAGIC TOADSTOOL        -5   
37      C536548     20914  SET/5 RED RETROSPOT LID GLASS BOWLS        -1   
38      C536548     22892    SET OF SALT AND PEPPER TOADSTOOLS        -7   
39      C536548     22654                   DELUXE SEWING KIT         -1   
...         ...       ...                                  ...       ...   
15089   C575064     22212           FOUR HOOK  WHITE LOVEBIRDS        -1   
15090   C575064     21669      BLUE STRIPE CERAMIC DRAWER KNOB        -1   
15091   C575064     20749           ASSORTED COLOUR MINI CASES        -1   
15092   C575064     20712           JUMBO BAG WOODLAND ANIMALS        -2   
15172   C575223     22973         CHILDREN'S CIRCUS PARADE MUG        -1   

               InvoiceDate  UnitPrice  CustomerID  Country  
35     2021-12-01 14:33:00       1.95     12472.0  Germany  
36     2021-12-01 14:33:00       1.65     12472.0  Germany  
37     2021-12-01 14:33:00       2.95     12472.0  Germany  
38     2021-12-01 14:33:00       1.25     12472.0  Germany  
39     2021-12-01 14:33:00       5.95     12472.0  Germany  
...                    ...        ...         ...      ...  
15089  2022-11-08 12:39:00       2.10     12472.0  Germany  
15090  2022-11-08 12:39:00       1.45     12472.0  Germany  
15091  2022-11-08 12:39:00       7.95     12472.0  Germany  
15092  2022-11-08 12:39:00       2.08     12472.0  Germany  
15172  2022-11-09 11:33:00       1.65     12472.0  Germany  

[391 rows x 8 columns]

We will check if the logic is correct

In [32]:
df[(df.CustomerID == 12472.0)&(df.StockCode == '20914')]

InvoiceNo StockCode                          Description  Quantity  \
37    C536548     20914  SET/5 RED RETROSPOT LID GLASS BOWLS        -1   
254    537201     20914  SET/5 RED RETROSPOT LID GLASS BOWLS         6   

             InvoiceDate  UnitPrice  CustomerID  Country  
37   2021-12-01 14:33:00       2.95     12472.0  Germany  
254  2021-12-05 14:19:00       2.95     12472.0  Germany

In [33]:
df[(df.CustomerID == 12472.0)&(df.StockCode == '22892')]

InvoiceNo StockCode                        Description  Quantity  \
38   C536548     22892  SET OF SALT AND PEPPER TOADSTOOLS        -7   

            InvoiceDate  UnitPrice  CustomerID  Country  
38  2021-12-01 14:33:00       1.25     12472.0  Germany

In [34]:
df[(df.CustomerID == 12472.0)&(df.StockCode == '20712')]

InvoiceNo StockCode                 Description  Quantity  \
3718     546843     20712  JUMBO BAG WOODLAND ANIMALS        10   
14087    571824     20712  JUMBO BAG WOODLAND ANIMALS        10   
15092   C575064     20712  JUMBO BAG WOODLAND ANIMALS        -2   

               InvoiceDate  UnitPrice  CustomerID  Country  
3718   2022-03-17 12:40:00       1.95     12472.0  Germany  
14087  2022-10-19 11:49:00       2.08     12472.0  Germany  
15092  2022-11-08 12:39:00       2.08     12472.0  Germany

In [35]:
df[(df.CustomerID == 12472.0)&(df.StockCode == '22244')]

InvoiceNo StockCode                 Description  Quantity  \
35     C536548     22244  3 HOOK HANGER MAGIC GARDEN        -4   
252     537201     22244  3 HOOK HANGER MAGIC GARDEN        12   
3722    546843     22244  3 HOOK HANGER MAGIC GARDEN        12   

              InvoiceDate  UnitPrice  CustomerID  Country  
35    2021-12-01 14:33:00       1.95     12472.0  Germany  
252   2021-12-05 14:19:00       1.95     12472.0  Germany  
3722  2022-03-17 12:40:00       1.95     12472.0  Germany

## Compartition of sales between Germany and France

In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18052 entries, 0 to 18051
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   InvoiceNo    18052 non-null  object 
 1   StockCode    18052 non-null  object 
 2   Description  18052 non-null  object 
 3   Quantity     18052 non-null  int64  
 4   InvoiceDate  18052 non-null  object 
 5   UnitPrice    18052 non-null  float64
 6   CustomerID   17986 non-null  float64
 7   Country      18052 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 1.1+ MB


In [45]:
df.describe()

Quantity     UnitPrice    CustomerID
count  18052.000000  18052.000000  17986.000000
mean      12.626191      4.470307  12661.178528
std       19.634816     56.311410    294.710582
min     -288.000000      0.000000  12413.000000
25%        5.000000      1.250000  12523.000000
50%       10.000000      1.950000  12625.000000
75%       12.000000      3.750000  12683.000000
max      912.000000   4161.060000  14335.000000

In [59]:
df['QpUP'] =  df.Quantity*df.UnitPrice

In [60]:
df

InvoiceNo StockCode                      Description  Quantity  \
0        536370     22728        ALARM CLOCK BAKELIKE PINK        24   
1        536370     22727        ALARM CLOCK BAKELIKE RED         24   
2        536370     22726       ALARM CLOCK BAKELIKE GREEN        12   
3        536370     21724  PANDA AND BUNNIES STICKER SHEET        12   
4        536370     21883                 STARS GIFT TAPE         24   
...         ...       ...                              ...       ...   
18047    581587     22613      PACK OF 20 SPACEBOY NAPKINS        12   
18048    581587     22899     CHILDREN'S APRON DOLLY GIRL          6   
18049    581587     23254    CHILDRENS CUTLERY DOLLY GIRL          4   
18050    581587     23255  CHILDRENS CUTLERY CIRCUS PARADE         4   
18051    581587     22138    BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID Country   QpUP  
0      2021-12-01 08:45:00       3.75     12583.0  France  90.00  
1      2021-12-01 08:45:00       3.75     12583.0  France  90.00  
2      2021-12-01 08:45:00       3.75     12583.0  France  45.00  
3      2021-12-01 08:45:00       0.85     12583.0  France  10.20  
4      2021-12-01 08:45:00       0.65     12583.0  France  15.60  
...                    ...        ...         ...     ...    ...  
18047  2022-12-09 12:50:00       0.85     12680.0  France  10.20  
18048  2022-12-09 12:50:00       2.10     12680.0  France  12.60  
18049  2022-12-09 12:50:00       4.15     12680.0  France  16.60  
18050  2022-12-09 12:50:00       4.15     12680.0  France  16.60  
18051  2022-12-09 12:50:00       4.95     12680.0  France  14.85  

[18052 rows x 9 columns]

In [61]:
df[df.Country =='France'].describe()

Quantity    UnitPrice    CustomerID         QpUP
count  8557.000000  8557.000000   8491.000000  8557.000000
mean     12.911067     5.028864  12677.995996    23.069288
std      21.425031    79.909126    276.742088   117.181844
min    -250.000000     0.000000  12413.000000 -8322.120000
25%       5.000000     1.250000  12571.000000    10.500000
50%      10.000000     1.790000  12674.000000    16.500000
75%      12.000000     3.750000  12689.000000    23.400000
max     912.000000  4161.060000  14277.000000  4161.060000

In [62]:
df[df.Country =='Germany'].describe()

Quantity    UnitPrice    CustomerID         QpUP
count  9495.000000  9495.000000   9495.000000  9495.000000
mean     12.369458     3.966930  12646.139336    23.348943
std      17.865719    16.549026    309.138487    37.324826
min    -288.000000     0.000000  12426.000000  -599.500000
25%       5.000000     1.250000  12480.000000    11.100000
50%      10.000000     1.950000  12592.000000    16.600000
75%      12.000000     3.750000  12662.000000    23.400000
max     600.000000   599.500000  14335.000000   876.000000

In [63]:
df[(df.Country =='France')&(df.Quantity <-100)]

InvoiceNo StockCode                          Description  Quantity  \
16688   C578377     22922     FRIDGE MAGNETS US DINER ASSORTED      -120   
16689   C578377     22923  FRIDGE MAGNETS LES ENFANTS ASSORTED      -120   
16690   C578377     22924        FRIDGE MAGNETS LA VIE EN ROSE      -120   
17110   C579192     22959               WRAP CHRISTMAS VILLAGE      -250   

               InvoiceDate  UnitPrice  CustomerID Country   QpUP  
16688  2022-11-24 11:44:00       0.72     14277.0  France  -86.4  
16689  2022-11-24 11:44:00       0.72     14277.0  France  -86.4  
16690  2022-11-24 11:44:00       0.72     14277.0  France  -86.4  
17110  2022-11-28 15:43:00       0.42     12657.0  France -105.0

In [64]:
df[(df.Country =='France')&(df.UnitPrice> 100)]

InvoiceNo StockCode Description  Quantity          InvoiceDate  \
918      539688        C2    CARRIAGE         1  2021-12-21 11:00:00   
4458     548725      POST     POSTAGE         1  2022-04-04 09:08:00   
4514     548913         M      Manual         1  2022-04-05 09:45:00   
4515    C548914         M      Manual        -1  2022-04-05 09:46:00   
8476     559889      POST     POSTAGE         1  2022-07-13 11:48:00   
10475    564862      POST     POSTAGE         1  2022-08-31 10:24:00   
14420    573077         M      Manual         1  2022-10-27 14:13:00   
14421   C573079         M      Manual        -2  2022-10-27 14:15:00   
14422    573080         M      Manual         1  2022-10-27 14:20:00   
14504    573153      POST     POSTAGE         3  2022-10-28 07:39:00   

       UnitPrice  CustomerID Country     QpUP  
918       150.00     12678.0  France   150.00  
4458      134.00     12678.0  France   134.00  
4514     1136.30     12669.0  France  1136.30  
4515     1136.30     12669.0  France -1136.30  
8476      145.00     12678.0  France   145.00  
10475     145.00     12678.0  France   145.00  
14420    4161.06     12536.0  France  4161.06  
14421    4161.06     12536.0  France -8322.12  
14422    4161.06     12536.0  France  4161.06  
14504     145.00     12678.0  France   435.00

In [65]:
df[(df.Country =='Germany')&(df.Quantity <-100)]

InvoiceNo StockCode                    Description  Quantity  \
17014   C579090     84598  BOYS ALPHABET IRON ON PATCHES      -288   

               InvoiceDate  UnitPrice  CustomerID  Country   QpUP  
17014  2022-11-28 10:48:00       0.21     12427.0  Germany -60.48

In [66]:
df[(df.Country =='Germany')&(df.UnitPrice> 100)]

InvoiceNo StockCode Description  Quantity          InvoiceDate  \
4857     550354         M      Manual         1  2022-04-18 10:28:00   
4858    C550355         M      Manual        -1  2022-04-18 10:29:00   
5757     552980      POST     POSTAGE         1  2022-05-12 14:52:00   
8045     558895         M      Manual         1  2022-07-04 15:54:00   
8046    C558897         M      Manual        -1  2022-07-04 15:55:00   
12686    569639      POST     POSTAGE         1  2022-10-05 12:24:00   
13703   C571220         M      Manual        -1  2022-10-14 13:35:00   
13704    571223         M      Manual         1  2022-10-14 13:36:00   
15247    575632         M      Manual         1  2022-11-10 13:44:00   
15248    575632         M      Manual         1  2022-11-10 13:44:00   
15249   C575635         M      Manual        -1  2022-11-10 13:45:00   
15250   C575635         M      Manual        -1  2022-11-10 13:45:00   
17863    581179      POST     POSTAGE         1  2022-12-07 15:43:00   

       UnitPrice  CustomerID  Country    QpUP  
4857      222.75     13811.0  Germany  222.75  
4858      222.75     13811.0  Germany -222.75  
5757      523.00     12590.0  Germany  523.00  
8045      389.68     12619.0  Germany  389.68  
8046      389.68     12619.0  Germany -389.68  
12686     240.00     12471.0  Germany  240.00  
13703     599.50     13810.0  Germany -599.50  
13704     599.50     13810.0  Germany  599.50  
15247     424.06     12473.0  Germany  424.06  
15248     549.34     12473.0  Germany  549.34  
15249     424.06     12473.0  Germany -424.06  
15250     549.34     12473.0  Germany -549.34  
17863     240.00     12471.0  Germany  240.00

In [69]:
df[df.Country == 'Germany'].QpUP.sum()

221698.21

Germany in total of Unit prices of sales is 221,698.21

In [70]:
df[df.Country == 'France'].QpUP.sum()

197403.9

France in total of Unit prices of sales is 197,403.9

In [71]:
df[df.Country == 'Germany'].Quantity.sum()

117448

Germany sold 117,448 products

In [72]:
df[df.Country == 'France'].Quantity.sum()

110480

France sold 110,480 products

### The dataset had some issues, with value outliers, that required further research.
### 
But we can still detect that Germany had more sales than France, but the French mean unit price is a bit higher. In general, the Germany branch had sold more of the QpUP, the recommendation would be to invest more in Germany compared with France.